tools/build-tools/src/typeValidator/testGeneration.ts

```diff
@@ -27,22 +27,38 @@ import * as current from "../index";
         for(const type of currentTypeData.typeData){
             const typeString = type.name.replace(".","");
             // no need to test new types
-            if(!type.internal && oldTypes.some((t)=>t.name.replace(".","") == typeString)){
+            if(oldTypes.some((t)=>t.name.replace(".","") == typeString)){
                 const oldType = `old.${type.name}`
                 const currentType = `current.${type.name}`
 
+                if(currentTypeData.packageDetails.broken[type.name]?.forwardCompat !== false){
```

Comment:
Actually adding these lines everywhere gets pretty verbose as seen in the example test case.  Seems like this might be better as a block at the top of the files?

Reply:
are you thinking the test file or the source file. i'm thinking i will need to incorporate version here as well

Reply:
this was about the test file like we discussed separately